### PR TITLE
NFT-345 feat: opengraph on loan page

### PR DESCRIPTION
**Pull request by cnasc**
see: https://www.opengraph.xyz/url/https%3A%2F%2Finterface-git-cjn-og-backed.vercel.app%2Fnetwork%2Fethereum%2Floans%2F63%2F/

### CODE REVIEW

File: pages/api/network/[network]/nftInfo/[contractAddress]/[tokenId].ts [hunk 1]
```diff
@@ -58,6 +45,38 @@ async function handler(
 
 export default withSentry(handler);
 
+/**
+ * Only for use on backend. When fetching NFT metadata from client, prefer
+ * hitting the api endpoint or using a function that calls it (such as
+ * `getNFTInfoFromTokenInfo`)
+ */
+export async function getMetadata(
```
Comment (adamgobes) [inline pages/api/network/[network]/nftInfo/[contractAddress]/[tokenId].ts:53]: so i wrote a method https://github.com/with-backed/backed-interface/blob/e9f497f12ce5a26f5864059497476b301f1c81d3/lib/events/consumers/getNftInfoForAttachment.ts#L7

used here: https://github.com/with-backed/backed-interface/blob/e9f497f12ce5a26f5864059497476b301f1c81d3/lib/events/consumers/discord/formatter.ts#L49

do you think we can consolidate? is there a reason this method does not make request to api endpoint?

Comment (cnasc) [reply pages/api/network/[network]/nftInfo/[contractAddress]/[tokenId].ts:53]: > is there a reason this method does not make request to api endpoint

@adamgobes Yes, you're not supposed to call API endpoints during SSR. You're on the server, so you just call the method. see: https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props#getserversideprops-or-api-routes

Comment (adamgobes) [reply pages/api/network/[network]/nftInfo/[contractAddress]/[tokenId].ts:53]: ahh gotcha!

**Review by adamgobes — APPROVED** [review]
:shipit:

**Pull request merged**